modules/python/nitf/source/nitro.i

```diff
@@ -570,15 +570,19 @@
         newWindow->numRows = window->numRows;
         newWindow->numCols = window->numCols;
         newWindow->numBands = 1;
-        
+
         result = PyList_New(window->numBands);
         for (i = 0; i < window->numBands; i++)
         {
             PyObject* buffObj = PyBuffer_New(subimageSize * sizeof(nitf_Uint8));
             if (!buffObj) goto CATCH_ERROR;
             buffObj->ob_type->tp_as_buffer->bf_getwritebuffer(buffObj, 0, (void **)&buf[0]);
-            
+
             newWindow->bandList = &window->bandList[i];
+
+            //We need to renumber the band going in to make a check pass,
+            //since we tell it that numBands = 1
+            newWindow->bandList[0] = 0;
```

Comment:
Just pushed up something that should work.

From: Adam Sylvester [mailto:notifications@github.com]
Sent: Monday, March 14, 2016 12:39 PM
To: mdaus/nitro <nitro@noreply.github.com>
Cc: Jonathan Means <Jonathan.Means@mdaus.com>
Subject: Re: [nitro] Nitropy fixes (#7)


In modules/python/nitf/source/nitro.i<http://cp.mcafee.com/d/5fHCN0e3zqb3UUQszCnSbCXCSm6hMUQsLLCTNPb3a9EVd79J6XRTD3hPsQsL6QkmkkTPhOUM-r4qMSpfosH-ME2BkdrFOVJl3mWsKrojd7fKuU_R-uojd7d-LsKCONvAhMUsyVORQX8EGTsjVkffGhBrwqrhdFCXYDuZXTLuZPtPpesRG9px6kOrp3BYlgRKCgBedsmrffMs1y4GmrFilo2InTbIEIEK6UlzWNVo54pj9JAendIcIICMnWhEwdbs9jzkQKCy06iD6GNoQgeO3_Eq83hg1wUQkCSrCVg3JIotBGQ15>:

>              newWindow->bandList = &window->bandList[i];

> +

> +            //We need to renumber the band going in to make a check pass,

> +            //since we tell it that numBands = 1

> +            newWindow->bandList[0] = 0;

I don't understand why this is needed... if you hardcode bandList to always be 0, won't you not read out the band you want? Seems like you'll read the same band each time through the loop. Or am I missing something?

—
Reply to this email directly or view it on GitHub<http://cp.mcafee.com/d/k-Kr6hEe6x0SyM-ed78VBZyVKVJBxAsed7bXVJYsOMOyqejhOrhKZtVMQsTd7bNJ55B5dYQsKcfCN6IdCjS7a_Ia0Fl3mWsKrlgRKDbCS4PhPXDKfZvDC4PhPvHTbFIInV4se78KsJteOaaJT4-l3PWApmU6CQPqpK_9TLuZXTLsTsSjDdqymohBcCSgVv5kdrFA9jzn5CPPY7oX8_W0p5ZOXababxK5o-Ium1h6kOrp3BPr3bb9I5-Aq83iT2kURdbFEw1AFNGImd43Iw_W6y0Qk0oed59JCVJuHmK1SQB57U>.


The information contained in this communication is confidential, is intended only for the use of the recipient(s) named above, and may be legally privileged. If the reader of this message is not the intended recipient, you are hereby notified that any dissemination, distribution, or copying of this communication is strictly prohibited. If you have received this communication in error, please re-send this communication to the sender and delete the original message or any copy of it from your computer system.